editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -86,66 +92,86 @@ export function setGridPropsCommands(
       PP.create('style', 'gridRowEnd'),
     ]),
   ]
-  const columnStart = gridPositionToValue(gridProps.gridColumnStart, null)
-  const columnEnd = gridPositionToValue(gridProps.gridColumnEnd, gridProps.gridColumnStart ?? null)
-  const rowStart = gridPositionToValue(gridProps.gridRowStart, null)
-  const rowEnd = gridPositionToValue(gridProps.gridRowEnd, gridProps.gridRowStart ?? null)
-
-  const lineColumnStart = asMaybeNamedLineOrValue(gridTemplate, 'column', columnStart)
-  const lineColumnEnd = asMaybeNamedLineOrValue(gridTemplate, 'column', columnEnd)
-  const lineRowStart = asMaybeNamedLineOrValue(gridTemplate, 'row', rowStart)
-  const lineRowEnd = asMaybeNamedLineOrValue(gridTemplate, 'row', rowEnd)
-
-  if (columnStart != null && columnStart === columnEnd) {
-    commands.push(
-      setProperty('always', elementPath, PP.create('style', 'gridColumn'), lineColumnStart),
-    )
-  } else if (
-    columnStart != null &&
-    typeof columnStart === 'number' &&
-    columnEnd != null &&
-    typeof columnEnd === 'number' &&
-    columnStart === columnEnd - 1
-  ) {
-    commands.push(
-      setProperty('always', elementPath, PP.create('style', 'gridColumn'), lineColumnStart),
-    )
-  } else {
-    if (columnStart != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridColumnStart'), lineColumnStart),
-      )
+
+  function printPin(pin: GridPositionOrSpan, axis: 'row' | 'column'): string | number {
+    if (isGridSpan(pin)) {
+      return stringifyGridSpan(pin)
     }
-    if (columnEnd != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridColumnEnd'), lineColumnEnd),
-      )
+    if (isCSSKeyword(pin)) {
+      return pin.value
     }
+    const tracks =
+      axis === 'column' ? gridTemplate.gridTemplateColumns : gridTemplate.gridTemplateRows
+    const maybeLineName =
+      tracks?.type === 'DIMENSIONS'
+        ? tracks.dimensions.find((_, index) => index + 1 === pin.numericalPosition)?.lineName
+        : null
+    if (maybeLineName != null) {
+      return maybeLineName
+    }
+    return pin.numericalPosition ?? 'auto'
   }
 
-  if (rowStart != null && rowStart === rowEnd) {
-    commands.push(setProperty('always', elementPath, PP.create('style', 'gridRow'), lineRowStart))
-  } else if (
-    rowStart != null &&
-    typeof rowStart === 'number' &&
-    rowEnd != null &&
-    typeof rowEnd === 'number' &&
-    rowStart === rowEnd - 1
-  ) {
-    commands.push(setProperty('always', elementPath, PP.create('style', 'gridRow'), lineRowStart))
-  } else {
-    if (rowStart != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridRowStart'), lineRowStart),
-      )
+  function serializeAxis(
+    startPosition: GridPositionOrSpan,
+    endPosition: GridPositionOrSpan,
+    axis: 'row' | 'column',
+  ): {
+    property:
+      | 'gridColumn'
+      | 'gridColumnStart'
```

Comment:
this and the other start are currently unused but I figured it's just easy to leave them here if we want to play around with this logic a bit